lib/openhab/core/types/string_type.rb

```diff
@@ -26,7 +26,7 @@ class StringType
         def eql?(other)
           return false unless other.instance_of?(self.class)
 
-          to_s.compare_to(other.to_s).zero?
+          to_s == other.to_s
```

Comment:
This should just be `equals(other)` (using the Java comparison, to avoid extraneous conversions to String)

Reply:
I tried that too. For some reason, equals against string returns true so I put this back in. Is there a Ruby version of equals that might have been called instead? The implementation of StringType.equals checks for the object's class so I don't get how it could return true for ruby string.